### PR TITLE
Add Beidou ephemeris message

### PR DESCRIPTION
**Pull request by Paakkit**
This PR adds Beidou SBP message.
Beidou has 2 tgd parameters in ephemeris, and thus cannot re-use GPS ephemeris message.

https://swift-nav.atlassian.net/projects/EX/issues/EX-45

### CODE REVIEW

**Comment by Paakkit** [comment]
First libsbp PR... let's see how it goes.

**Comment by Paakkit** [comment]
Lots of auto generated changes...

File: javascript/sbp.bundle.js [hunk 1]
```diff
@@ -15817,65 +15915,97 @@ for (var i = 0, len = code.length; i < len; ++i) {
 revLookup['-'.charCodeAt(0)] = 62
 revLookup['_'.charCodeAt(0)] = 63
 
-function placeHoldersCount (b64) {
+function getLens (b64) {
```
Comment (Paakkit) [inline javascript/sbp.bundle.js]: I wonder what are these changes, starting from here until the end of this file?
They don't seem to be related to the new message.

**Comment by mbavaro** [comment]
Beidou eph significant bits:
image


GPS
image


Galileo
image

File: spec/yaml/swiftnav/sbp/observation.yaml [hunk 2]
```diff
@@ -445,6 +445,106 @@ definitions:
...
+      The ephemeris message returns a set of satellite orbit
+      parameters that is used to calculate BDS satellite position,
+      velocity, and clock offset. Please see the BeiDou Navigation
+      Satellite System SIS-ICD Version 2.1, Table 5-9 for more details.
+    fields:
+      - common:
+          type: EphemerisCommonContent
+          desc: Values common for all ephemeris types
+      - tgd:
+          type: array
```
Comment (mfine) [inline spec/yaml/swiftnav/sbp/observation.yaml]: Does the produced ephemeris use an array for these values as well? Or would it be clearer to break out the two values into distinct fields. I don't love using arrays to group items where distinct fields can be used.

**Comment by mbavaro** [comment]
also, there is a reason why GNSS send semi-circles and not radians.. that is not to embed the definition of `PI` in the parameter.. not for this PR because it would have a massive impact on LSNP too, but eventually might make sense to do the same and leave `PI` to the user

**Comment by Paakkit** [comment]
pdf contained question marks.
Doc generation needed to be run twice to resolve Latex links.

**Comment by Paakkit** [comment]
@mbavaro I'll update this PR to use floats for Beidou.
Let's consider a new PR for changing the existing messages.
Also, I'd consider this a good moment to introduce the Galileo message.

**Comment by Paakkit** [comment]
I spoke too soon about the floats - array of floats gives an error.
The array of doubles is in https://github.com/swift-nav/libsbp/blob/master/java/src/com/swiftnav/sbp/SBPMessage.java, but floats are not handled.

Having `float tgd1` and `float tgd2` would work fine.

**Comment by mbavaro** [comment]
I insist... I think adding `getArrayofFloat()` like this:
https://github.com/swift-nav/libsbp/blob/master/java/src/com/swiftnav/sbp/SBPMessage.java#L159-L164
should be possible.

SBP has some contradictions.. it's designed for minimal payloads (so much that messages are constrained to 255 Bytes) but then our ephemeris messages are completely bloated. Giving it a quick glance, full use of `float`s here should reduce the bandwidth of that message to around 67% of the original, and show to the GNSS community that we know what we are doing and aren't just using `double`s to be on the safe side :)

Happy to hear about your Galileo proposition, let me know if you want to take it or I should.

File: spec/yaml/swiftnav/sbp/observation.yaml (hunk 2 above)
Comment (anth-cole) [reply spec/yaml/swiftnav/sbp/observation.yaml]: I'd prefer to break away from the array in the SBP message I think - in this instance it'll be much clearer and easier to understand

**Comment by anth-cole** [comment]
so I'll chime in - I absolutely do not want to have the situation where we have BDS ephemeris (and GAL) using floats and the GPS and GAL ones using doubles. Talking to @denniszollo this morning, he suggested if we're trying to optimise bandwidth, is it worth considering going to the raw nav bits? @mbavaro thoughts on this approach? I'm worried about needing to implement an ephemeris decoder everywhere, but it would optimise out throughput.

With the question of floats, I'd either like to convert GPS and GLO messages first and propagate that change through, then add consistent BDS+GAL eph messages, or discuss the idea of adding raw nav messages. I very much agree the current messages are bloated and cause us to have bandwidth spikes.

File: spec/yaml/swiftnav/sbp/observation.yaml (hunk 2 above)
Comment (Paakkit) [reply spec/yaml/swiftnav/sbp/observation.yaml]: The array has been removed.

**Comment by Paakkit** [comment]
This PR has been updated with the suggested changes.
In the meanwhile I'll open a new PR for applying the float changes to the existing ephemeris messages.

**Comment by Paakkit** [comment]
Rebased, and now Travis passes again.

**Comment by anth-cole** [comment]
lgtm - please merge

**Comment by mbavaro** [comment]
@Paakkit I merged the Galileo and broke yours, which I now rebased so it should be good to put in when you're back in the office
